analyses/org.eclipse.tracecompass.incubator.executioncomparision.core/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/core/DifferentialSeqCallGraphAnalysis.java

```diff
@@ -0,0 +1,409 @@
+     *
+     */
+    @TmfSignalHandler
+    public void selectionRangeUpdated(TmfComparisonFilteringUpdatedSignal signal) {
+        // tuning start and end times
+        fStartA = TmfTimestamp.BIG_BANG.equals(signal.getBeginTimeA()) ? fStartA : signal.getBeginTimeA();
+        fEndA = TmfTimestamp.BIG_CRUNCH.equals(signal.getEndTimeA()) ? fEndA : signal.getEndTimeA();
+        fStartB = TmfTimestamp.BIG_BANG.equals(signal.getBeginTimeB()) ? fStartB : signal.getBeginTimeB();
+        fEndB = TmfTimestamp.BIG_CRUNCH.equals(signal.getEndTimeB()) ? fEndB : signal.getEndTimeB();
+        // tuning fStatistic
+        String statistic = signal.getStatistic();
+        fStatistic = (statistic == null) ? fStatistic : statistic;
+        // tuning fTraceList
+        List<String> traceListA = signal.getTraceListA();
+        if (traceListA != null) {
+            synchronizedListAdd(traceListA, fTraceListA);
+
+        }
+        List<String> traceListB = signal.getTraceListB();
+        if (traceListB != null) {
+            synchronizedListAdd(traceListB, fTraceListB);
+
+        }
+        if (!fTraceCallGraphRegistry.isEmpty()) {
+            try (ScopeLog sl = new ScopeLog(LOGGER, Level.FINE, "MakeDiffCallGraph")) { //$NON-NLS-1$
+                synchronized (this) {
+                    if (fDiffJob != null) {
+                        fDiffJob.cancel();
+                    }
+                    fDiffJob = new Job("Make differential Callgraph") { //$NON-NLS-1$
+                        @Override
+                        protected IStatus run(@Nullable IProgressMonitor monitor) {
+                            refreshDiffCG(monitor);
+                            if (monitor != null) {
+                                monitor.done();
+                            }
+                            return Status.OK_STATUS;
+                        }
+                    };
+                    fDiffJob.schedule();
+                }
+            }
+        }
+    }
+
+    /**
+     * Helper function that adds all elements from the given traceList to the
+     * given fTraceList in a thread-safe manner.
+     *
+     * @param traceList
+     *            the list of strings to be added
+     * @param fTraceList
+     *            the list to which the strings will be added
+     * @return the synchronized list after adding the elements
+     */
+    private static List<String> synchronizedListAdd(List<String> traceList, List<String> fTraceList) {
+        List<String> synchronizedList = Collections.synchronizedList(fTraceList);
+        synchronized (synchronizedList) {
+            synchronizedList.clear();
+            for (String name : traceList) {
+                synchronizedList.add(name);
+            }
+        }
+        return synchronizedList;
+    }
+
+    @Override
+    public boolean canExecute(ITmfTrace trace) {
+        return (trace instanceof TmfExperiment);
+
+    }
+
+    @Override
+    public void dispose() {
```

Comment:
This method should cancel all running jobs that are running outside of the executeAnalysis method or that call directly executeAnalysis(). See the implementation of super.dispose() for more details.